### DISCUSSION TITLE
Create Loop-Device for complex Image

### DISCUSSION THREAD

**Discussion by killermoehre** [systemd/mkosi#2413] [Q&A] [answered]
Hi,

I need to create my own disk image with a base in LVM for real hardware machines. As `systemd-repart` has no support for this, I create my own disk in `mkosi.finalize` and give it an appropriate GPT with `sgdisk`, to be a `rsync` target of `$BUILDROOT` after mounting all the partitions.

Now the problem is that I can't loop-mount this disk with `losetup`, because in the `mkosi.finalize` name space there are no loop devices (`ls /dev/loop*` is empty).

Even if I `mknod -m0777 "$LOOP_DEVICE" b 7 0; losetup "$LOOP_DEVICE" "$RAW_DISK"` I get `Permission denied` inside `mkosi.finalize`.

Any idea? `mkosi.prepare` is probably already to late for this setup. Also, this setup worked with `mkosi@v14`.

**Comment by killermoehre** [comment]
Hmm … https://github.com/containers/bubblewrap/issues/523 looks unfortunate in this regard. Seems like loop devices are _not_ namespace'able. I'm open for suggestions on how to solve this within `mkosi`.

**Comment by daandemeyer** [comment, marked as answer]
@killermoehre This is never going to work within mkosi. See `test_initrd_lvm()` in `test_initrd.py` for how we test LVM. Just write a wrapper script that builds the image and then copies the result over into the LVM stuff.

**Reply by killermoehre** [reply]
Bummer. But ok. So this will become a multistep setup. Thanks for the heads-up.
